dipy/tracking/utils.py

```diff
@@ -595,6 +596,45 @@ def target(streamlines, target_mask, affine, include=True):
             yield sl
 
 
+@_with_initialize
+def target_line_based(streamlines, target_mask, affine=None, include=True):
+    """Filters streamlines based on whether or not they pass through an ROI,
+    using a line-based algorithm for compressed tract.
```

Comment:
I know that this is for now the most frequent use case, but it's useful for any tractogram. For example, the step size might be "big" (eg: 0.5mm). In this case, this function would still give more robust results, since we get even the corner of voxels traversed by a streamline, even if no point is inside the voxel.

Will also be useful for tractograms where the step size is varying, eg: global tractography results.